spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/dsl/StoredProcExecutorConfigurer.java

```diff
@@ -0,0 +1,163 @@
+		this.storedProcExecutor.setSqlParameters(sqlParameters);
+		return this;
+	}
+
+	/**
+	 * @param sqlParameterSourceFactory the sqlParameterSourceFactory
+	 * @return the storedProcExecutor
+	 * @see StoredProcExecutor#setSqlParameterSourceFactory(SqlParameterSourceFactory)
+	 */
+	public StoredProcExecutorConfigurer sqlParameterSourceFactory(SqlParameterSourceFactory sqlParameterSourceFactory) {
+		this.storedProcExecutor.setSqlParameterSourceFactory(sqlParameterSourceFactory);
+		return this;
+	}
+
+	/**
+	 * @param storedProcedureName the storedProcedureName
+	 * @return the storedProcExecutor
+	 * @see StoredProcExecutor#setStoredProcedureName(String)
+	 */
+	public StoredProcExecutorConfigurer storedProcedureName(String storedProcedureName) {
+		this.storedProcExecutor.setStoredProcedureName(storedProcedureName);
+		return this;
+	}
+
+	/**
+	 * @param storedProcedureNameExpression the storedProcedureNameExpression
+	 * @return the storedProcExecutor
+	 * @see StoredProcExecutor#setStoredProcedureNameExpression(Expression)
+	 */
+	public StoredProcExecutorConfigurer storedProcedureNameExpression(Expression storedProcedureNameExpression) {
+		this.storedProcExecutor.setStoredProcedureNameExpression(storedProcedureNameExpression);
+		return this;
+	}
+
+	/**
+	 * @param usePayloadAsParameterSource the usePayloadAsParameterSource
+	 * @return the storedProcExecutor
+	 * @see StoredProcExecutor#setUsePayloadAsParameterSource(boolean)
+	 */
+	public StoredProcExecutorConfigurer usePayloadAsParameterSource(boolean usePayloadAsParameterSource) {
+		this.storedProcExecutor.setUsePayloadAsParameterSource(usePayloadAsParameterSource);
+		return this;
+	}
+
+	/**
+	 * @param isFunction the isFunction
+	 * @return the storedProcExecutor
+	 * @see StoredProcExecutor#setIsFunction(boolean)
+	 */
+	public StoredProcExecutorConfigurer isFunction(boolean isFunction) {
+		this.storedProcExecutor.setIsFunction(isFunction);
+		return this;
+	}
+
+	/**
+	 * @param returnValueRequired the returnValueRequired
+	 * @return the storedProcExecutor
+	 * @see StoredProcExecutor#setReturnValueRequired(boolean)
+	 */
+	public StoredProcExecutorConfigurer returnValueRequired(boolean returnValueRequired) {
+		this.storedProcExecutor.setReturnValueRequired(returnValueRequired);
+		return this;
+	}
+
+	/**
+	 * @param skipUndeclaredResults the skipUndeclaredResults
+	 * @return the storedProcExecutor
+	 * @see StoredProcExecutor#setSkipUndeclaredResults(boolean)
+	 */
+	public StoredProcExecutorConfigurer skipUndeclaredResults(boolean skipUndeclaredResults) {
+		this.storedProcExecutor.setSkipUndeclaredResults(skipUndeclaredResults);
+		return this;
+	}
+
+	/**
+	 * @param returningResultSetRowMappers the returningResultSetRowMappers
+	 * @return the storedProcExecutor
+	 * @see StoredProcExecutor#setReturningResultSetRowMappers(Map)
+	 */
+	public StoredProcExecutorConfigurer returningResultSetRowMappers(Map<String, RowMapper<?>> returningResultSetRowMappers) {
```

Comment:
Same idea here:
```
returningResultSetRowMapper(String name, RowMapper<?> rowMapper)
```
Then we build map internally.
(Same for the mentioned `procedureParameters` list to be built internally).

My point is that Java DSL should be as smooth as possible without extra hops.